framework/structures.md

```diff
@@ -118,70 +122,78 @@ The encryption context MUST reserve the following key fields for use by the AWS
 The encryption context SHOULD reserve any key field with the prefix `aws` for use by AWS KMS and
 other AWS services.
 
-### Encryption Materials
+### Data Key Materials
 
 #### Implementations
 
-- [Java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/model/EncryptionMaterials.java)
-- [Python](https://github.com/aws/aws-encryption-sdk-python/blob/master/src/aws_encryption_sdk/materials_managers/__init__.py)
-- [C](https://github.com/aws/aws-encryption-sdk-c/blob/master/include/aws/cryptosdk/materials.h)
-- [Javascript](https://github.com/awslabs/aws-encryption-sdk-javascript/blob/master/modules/material-management/src/cryptographic_material.ts)
+- [Java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/DataKey.java)
 
-#### Structure
+TODO: The above is only a similar data structure related to the deprecated master key/master key provider concepts.
 
-Encryption materials are a structure containing materials needed for [encryption](#encrypt.md).
-This structure MAY include any of the following fields:
+##### Algorithm Suite
 
-- [Algorithm Suite](#algorith-suite)
-- [Encrypted Data Keys](#encrypted-data-keys)
-- [Encryption Context](#encryption-context)
-- [Keyring Trace](#keyring-trace)
-- [Plaintext Data Key](#plaintext-data-key)
-- [Signing Key](#signing-key)
+The [algorithm suite](#algorithm-suites.md) to be used for [encryption](#encrypt.md) or [decryption](#decrypt.md).
 
-##### Algorithm Suite
+##### Plaintext Data Key
+
+A data key to be used as input for [encryption](#encrypt.md).
+
+The plaintext data key MUST:
+
+- fit the specification for the [key derivation algorithm](#algorithm-suites.md#key-derivation-algorithm)
+  included in this decryption material's [algorithm suite](#algorithm-suite).
+- consist of cryptographically secure (pseudo-)random bits.
+- be kept secret.
+
+The plaintext data key SHOULD be stored as immutable data.
```

Comment:
Why is the data structure being immutable important here? What security properties does that add?

I would argue that we don't really care about the mutability of the data structure; we care about the ability for the language/runtime/os/etc to manage and destroy the contents of that data structure.

Again, I specifically remember a lengthy discussion around the requirements for plaintext data keys...